core/src/main/java/org/infinispan/globalstate/ScopedPersistentState.java

```diff
@@ -27,6 +27,26 @@
    String getProperty(String key);
 
    /**
+    * Sets an integer state property.
+    */
+   void setProperty(String key, int value);
+
+   /**
+    * Sets a float state property.
+    */
+   void setProperty(String format, float f);
+
+   /**
+    * Retrieves an integer state property
+    */
+   int getIntProperty(String key);
```

Comment:
Not sure you need these. You could have just defined:

```
<T> T getProperty(String key);
```

And that'd cover all types...

Regardless of the API, the possibility of class cast exceptions remains with a system-property-like approach :|